nilearn/regions/signal_extraction.py

```diff
@@ -427,12 +444,58 @@ def img_to_signals_maps(imgs, maps_img, mask_img=None):
     use_mask = _check_shape_and_affine_compatibility(imgs, mask_img)
     if use_mask:
         mask_img = _utils.check_niimg_3d(mask_img)
+        labels_before_mask = set(labels)
         maps_data, maps_mask, labels = _trim_maps(
             maps_data,
             _safe_get_data(mask_img, ensure_finite=True),
-            keep_empty=True,
+            keep_empty=keep_masked_maps,
         )
         maps_mask = _utils.as_ndarray(maps_mask, dtype=bool)
+        if keep_masked_maps:
+            warnings.warn(
+                'Starting in version 0.15, the behavior of "NiftiMapsMasker" '
+                'will change when a mask is supplied through the "mask_img" '
+                'parameter. The atlases are masked by "mask_img" before any '
+                'signal extraction happens. However, some maps in the atlas '
+                'may contain no brain coverage after applying the mask, '
+                'resulting in an invalid map with only zeroes (not suitable '
+                'for signal extraction). These invalid maps used to be kept. '
+                'In the new behavior, they will be removed from the output. '
+                '\n\n'
+                'If "keep_masked_maps" is set to True, the masked atlas with '
+                'these invalid maps will be retained in the output, resulting '
+                'in corresponding time series with zeros only (old behavior). '
+                'To enable this '
+                'behavior, specify the parameter "keep_masked_maps=True" when '
+                'initializing the "NiftiMapsMasker" object.\n\n'
+                'Starting from version 0.13, the default behavior will be '
+                'changed to "keep_masked_maps=False". If "keep_masked_maps" '
+                'is set to False, the invalid maps will be removed from the '
+                'trimmed atlas, ensuring no empty time series are present in '
+                'the output (new behavior). '
+                'To explicitly disable the retention of masked '
+                'maps, specify the parameter "keep_masked_maps=False" when '
+                'initializing the "NiftiMapsMasker" object.'
+                '"keep_masked_maps" parameter will be removed '
+                'in version 0.15.',
```

Comment:
This is way too long for a deprecation warning.

I would just state three things

1. The output will contain empty time series.
2. Starting from version 0.13, the default behavior will be `keep_masked_maps=False`
3. `keep_masked_maps` will be removed in version 0.15

Just point the users to read the docs.

Probably tighten this up and put it in the change log?